### PR TITLE
Split nt.go into multiple files

### PR DESCRIPTION
**Pull request by karlkfi**
Group methods by domain:
- wait
- wait-for-sync
- validate
- metrics

### CODE REVIEW

File: e2e/nomostest/wait-for-sync.go [hunk 1]
```diff
@@ -0,0 +1,482 @@
+// Copyright 2022 Google LLC
```
Comment (sdowell) [inline e2e/nomostest/wait-for-sync.go:1]: Naming convention for files is snake case, right? Ditto with `config-sync.go`

Comment (karlkfi) [reply e2e/nomostest/wait-for-sync.go:1]: config-sync.go was already there. I just followed the package precedence, but I can rename them both.

Comment (karlkfi) [reply e2e/nomostest/wait-for-sync.go]: done

**Pull request merged**